core/src/View/Kokkos_BasicView.hpp

```diff
@@ -0,0 +1,611 @@
+      return map(indices[Idxs]...);
+    }(std::make_index_sequence<rank()>()));
+  }
+#endif
+
+  // C++20 operator()
+  template <class... OtherIndexTypes>
+    requires((std::is_convertible_v<OtherIndexTypes, index_type> && ...) &&
+             (std::is_nothrow_constructible_v<index_type, OtherIndexTypes> &&
+              ...) &&
+             (sizeof...(OtherIndexTypes) == rank()))
+  KOKKOS_FUNCTION constexpr reference operator()(
+      OtherIndexTypes... indices) const {
+    return acc.access(ptr, map(static_cast<index_type>(std::move(indices))...));
+  }
+
+  template <class OtherIndexType>
+    requires(
+        std::is_convertible_v<const OtherIndexType &, index_type> &&
+        std::is_nothrow_constructible_v<index_type, const OtherIndexType &>)
+  KOKKOS_FUNCTION constexpr reference operator()(
+      const Array<OtherIndexType, rank()> &indices) const {
+    return acc.access(ptr, [&]<size_t... Idxs>(std::index_sequence<Idxs...>) {
+      return map(indices[Idxs]...);
+    }(std::make_index_sequence<rank()>()));
+  }
+
+  template <class OtherIndexType>
+    requires(
+        std::is_convertible_v<const OtherIndexType &, index_type> &&
+        std::is_nothrow_constructible_v<index_type, const OtherIndexType &>)
+  KOKKOS_FUNCTION constexpr reference operator()(
+      std::span<OtherIndexType, rank()> indices) const {
+    return acc.access(ptr, [&]<size_t... Idxs>(std::index_sequence<Idxs...>) {
+      return map(indices[Idxs]...);
+    }(std::make_index_sequence<rank()>()));
+  }
+#else
+  // C++17 variant of operator()
+  template <class... OtherIndexTypes>
+  KOKKOS_FUNCTION constexpr std::enable_if_t<
+      (std::is_convertible_v<OtherIndexTypes, index_type> && ...) &&
+          (std::is_nothrow_constructible_v<index_type, OtherIndexTypes> &&
+           ...) &&
+          (sizeof...(OtherIndexTypes) == rank()),
+      reference>
+  operator()(OtherIndexTypes... indices) const {
+    return acc.access(ptr, map(static_cast<index_type>(std::move(indices))...));
+  }
+#endif
+
+ private:
+  // could use inline templated lambda in C++20 mode inside size()
+  template <size_t... Idxs>
+  KOKKOS_FUNCTION constexpr size_type size_impl(
+      std::index_sequence<Idxs...>) const noexcept {
+    return ((static_cast<size_type>(map.extents().extent(Idxs))) * ... *
+            size_type(1));
+  }
+
+ public:
+  KOKKOS_FUNCTION constexpr size_type size() const noexcept {
+    return size_impl(std::make_index_sequence<rank()>());
+  }
+
+ private:
+  // could use inline templated lambda in C++20 mode inside empty()
+  template <size_t... Idxs>
+  KOKKOS_FUNCTION constexpr bool empty_impl(
+      std::index_sequence<Idxs...>) const noexcept {
+    return (rank() > 0) &&
+           ((map.extents().extent(Idxs) == index_type(0)) || ... || false);
+  }
+
+ public:
+  [[nodiscard]] KOKKOS_FUNCTION constexpr bool empty() const noexcept {
+    return empt_impl(std::make_index_sequence<rank()>());
```

Comment:
```suggestion
    return empty_impl(std::make_index_sequence<rank()>());
```

Reply:
I guess I should add at least a simple call to empty in the tests ...

Reply:
fixed